hook/type.go

```diff
@@ -0,0 +1,54 @@
+package hook
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/buildkite/agent/v3/shellscript"
+)
+
+const (
+	TypeShell   = "shell"
+	TypeBinary  = "binary"
+	TypeScript  = "script" // Something with a shebang that isn't a shell script, but is an interpretable something-or-other
+	TypeUnknown = "unknown"
+)
+
+func Type(path string) (string, error) {
```

Comment:
It might be better to call this `FromPath`. Writing code like `type = type.Type` triggers me back to when I was forced to learn Java in Uni.

Reply:
i mean in this case it would be `type := hook.Type`, but i do see where you're coming from

Reply:
Whoops, yeah, `hook.Type` is fine.